### DISCUSSION TITLE
The issue of importing thermal data in the context of system coupling with PyMechanical.

### DISCUSSION THREAD

**Discussion by youthkiss** [ansys/pymechanical#1010] [Q&A] [answered]
I am attempting to perform an analysis using both the statics and steady-state thermal modules. 

First, I calculate the surface temperature of the optical components using the steady-state thermal module, and then import the temperature data into the statics module for surface deformation analysis. 

However, I am currently unable to carry out the calculation using this approach (①), and an error occurs when importing the temperature data into the statics module.
![PixPin_2024-12-26_16-41-54](https://github.com/user-attachments/assets/b60abb52-0bfb-49cd-ab3a-47972918ff84)

I would like to achieve the result shown in the following diagram. However, despite reviewing numerous example files, I have been unable to complete the above steps. I have attempted both the embedding and remote methods for the calculation, but neither approach has resolved the issue.
![PixPin_2024-12-26_16-43-46](https://github.com/user-attachments/assets/c40e65b5-663d-4c91-b337-0abc64311872)

Is there anything else I can do to help move this progress forward?

**Comment by youthkiss** [comment]
Additionally, I would like to ask how to import external data (e.g., .txt file) as Heat Flux and apply it to a specific surface.
[Uploading taper_1nm_Power.txt…]()

**Comment by dipinknair** [comment]
@youthkiss  Thanks for opening the discussion. Can you share the error message or logs that you are seeing with pymechanical ? 
If possible, can you share the script and other files needed to run? You can send them over [pyansys.core@ansys.com](mailto:pyansys.core@ansys.com) with title "pymechanical #1010".

**Reply by youthkiss** [reply, marked as answer]
@dipinknair 
Hi, recently, several methods have been identified to address the previously mentioned issues. Regarding the thermal-structural coupling aspect, calculations were performed using two scripts. The first script computes the steady-state thermal field of the components of interest, while the second script conducts structural calculations, effectively resolving the problem.
Concerning the issue of importing external data, the Scripting feature in the Mechanical GUI was utilized to record the operations, enabling the completion of the import through relevant code prompts.
Most importantly, it was discovered that parallel kernel crashes during the computation may occur due to data or other reasons. This issue can be resolved by adding the line <os.environ['KMP_DUPLICATE_LIB_OK'] = 'TRUE'> to the global variables, thereby preventing the crashes.

**Comment by dipinknair** [comment]
Ref #998
